plugins/hls-selection-range-plugin/src/Ide/Plugin/SelectionRange.hs

```diff
@@ -0,0 +1,144 @@
+                                                          Range (Range),
+                                                          fromNormalizedFilePath,
+                                                          ideLogger, logInfo,
+                                                          realSrcSpanToRange,
+                                                          runIdeAction,
+                                                          toNormalizedFilePath',
+                                                          uriToFilePath')
+import           Development.IDE.Core.Actions            (useE)
+import           Development.IDE.Core.PositionMapping    (PositionMapping,
+                                                          fromCurrentPosition,
+                                                          toCurrentRange)
+import           Development.IDE.GHC.Compat              (HieAST (Node), Span,
+                                                          getAsts)
+import           Development.IDE.GHC.Compat.Util
+import           Ide.Plugin.SelectionRange.ASTPreProcess (PreProcessEnv (PreProcessEnv),
+                                                          preProcessAST)
+import           Ide.PluginUtils                         (response)
+import           Ide.Types                               (PluginDescriptor (pluginHandlers),
+                                                          PluginId,
+                                                          defaultPluginDescriptor,
+                                                          mkPluginHandler)
+import           Language.LSP.Server                     (LspM)
+import           Language.LSP.Types                      (List (List),
+                                                          NormalizedFilePath,
+                                                          Position,
+                                                          ResponseError,
+                                                          SMethod (STextDocumentSelectionRange),
+                                                          SelectionRange (..),
+                                                          SelectionRangeParams (..),
+                                                          TextDocumentIdentifier (TextDocumentIdentifier),
+                                                          Uri)
+import           Prelude                                 hiding (span)
+
+descriptor :: PluginId -> PluginDescriptor IdeState
+descriptor plId = (defaultPluginDescriptor plId)
+    { pluginHandlers = mkPluginHandler STextDocumentSelectionRange selectionRangeHandler
+    }
+
+selectionRangeHandler :: IdeState -> PluginId -> SelectionRangeParams -> LspM c (Either ResponseError (List SelectionRange))
+selectionRangeHandler ide _ SelectionRangeParams{..} = do
+    liftIO $ logInfo logger $ "requesting selection range for file: " <> T.pack (show uri)
```

Comment:
Should this be `logDebug`?

Reply:
Emm, yep, this may not be appropriate to place in info log